libethash-cl/ethash_cl_miner.cpp

```diff
@@ -126,103 +127,166 @@ void ethash_cl_miner::finish()
 {
 	if (m_queue())
 		m_queue.finish();
+
+	if (m_dagChunks)
+		delete [] m_dagChunks;
```

Comment:
If  finish can be called twice m_dagChunks should be set to nullptr here. Right now finsh is public, so I guess this is possible